contracts/pairing.scrypt

```diff
@@ -0,0 +1,66 @@
+import "bn128.scrypt";
+
+
+library Pairing {
+
+    // Log of ATE_LOOP_COUNT
+    static const int N = 63;
+    static const bytes mask = reverseBytes(num2bin(1, N), N);
+
+    static const int ATE_LOOP_COUNT = 29793968203157093288;
+    static const bytes ATE_LOOP_COUNT_BYTES = reverseBytes(num2bin(1, ATE_LOOP_COUNT), ATE_LOOP_COUNT);
```

Comment:
Compute this off chain and set it. Currently, it generates over ATE_LOOP_COUNT opcodes.

Same above. Precompute off chain for constants known at compile time as much as possible.